File: Top_K_Frequent_Elements/Top_K_Frequent_Elements.cpp
class Solution
{
public:
    vector<int> topKFrequent(vector<int> &nums, int k)
    {
        unordered_map<int, int> hashmap;
        vector<vector<int>> frequency(nums.size() + 1);
        for (int n : nums)
        {
            hashmap[n]++;
        }

        for (auto &[key, value] : hashmap)
        {
            frequency[value].push_back(key);
        }

        vector<int> result;
        for (int i = nums.size(); i > 0; i--)
        {
            for (int n : frequency[i])
            {
                result.push_back(n);
                if (result.size() == k)
                {
                    return result;
                }
            }
        }
        return result;
    }
};

File: Top_K_Frequent_Elements/Top_K_Frequent_Elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        frequency = [[] for i in range(len(nums)+1)]
        for n in nums:
            hashmap[n] = 1 + hashmap.get(n, 0)

        for key, value in hashmap.items():
            frequency[value].append(key)

        result = []
        for i in range(len(nums), 0, -1):
            for n in frequency[i]:
                result.append(n)
                if (len(result) == k):
                    return result
